component/otelcol/extension/jaeger_remote_sampling/internal/jaegerremotesampling/extension.go

```diff
@@ -73,11 +73,7 @@ func (jrse *jrsExtension) Start(ctx context.Context, host component.Host) error
 	}
 
 	if jrse.cfg.Source.Remote != nil {
-		opts, err := jrse.cfg.Source.Remote.ToDialOptions(host, jrse.telemetry)
-		if err != nil {
-			return fmt.Errorf("error while setting up the remote sampling source: %w", err)
-		}
-		conn, err := grpc.Dial(jrse.cfg.Source.Remote.Endpoint, opts...)
+		conn, err := jrse.cfg.Source.Remote.ToClientConn(context.Background(), host, jrse.telemetry)
```

Comment:
The reason I used `context.Background()` is that I thought it's more in line with the previous implementation. Previously we'd [call](https://github.com/grafana/agent/blob/v0.34.1/component/otelcol/extension/jaeger_remote_sampling/internal/jaegerremotesampling/extension.go#L80) grpc.Dial, which [uses](https://github.com/grpc/grpc-go/blob/v1.52.3/clientconn.go#L104) the background context.

I think in this case the context doesn't matter, because the gRPC call is non-blocking and [DialContext](https://pkg.go.dev/google.golang.org/grpc@v1.55.0#DialContext) wouldn't need it.

If we pass the context we might confuse people who work on this in the future, because they might think it's important - that's why I omitted it. But if you think it's a best practice to always pass a context downstream if the current function has one, then I could pass it here too?

Reply:
This package is a fork from upstream, and upstream also [passes](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/c404e07bf35dd76aabe6a17f63a41850f782ce2b/extension/jaegerremotesampling/extension.go#L63) the context here, so we should too IMO.

Reply:
Good point, I'll update it!